### PR TITLE
changed many things (again)

### PR DESCRIPTION
**Pull request by StrawWagen**
buffed turret_grenade's projectile ( turret_40mm_frag ), blast damage from 45 to 90, blast range 400 to 350

reworked turret_bullet2 a bit, changed bullet damage to 70, and added a 20 damage explosion on bullet hit.
turret_bullet2 also creates a decal to communicate that yes, it does explode

added some comments that explain the variable "concrete"
also did heavy formatting in both projectile's code, everything should be less shit, and overall easier to read while being consistent. 
( doesn't help pr readability sadly )

fixed small bug as well, where blast damage was set up wrong for the two turret projectiles.
now, the shooter should inflict damage with the turret, instead of the turret inflicting damage with the shooter.

### CODE REVIEW

File: lua/entities/rail_shell/init.lua [hunk 1]
```diff
@@ -3,106 +3,112 @@ AddCSLuaFile( "shared.lua" )
 include('shared.lua')
```
Comment (StrawWagen) [inline lua/entities/rail_shell/init.lua:3]: only formatting for this file

File: lua/entities/rail_shell/init.lua [hunk 2]
```diff
@@ -3,106 +3,112 @@ AddCSLuaFile( "shared.lua" )
...
+			util.Effect( "watersplash", effectdata )
+			self.Entity:Remove()
+			return true
+			
+		end
+		
+		local owner = IsValid( self:GetOwner() ) and self:GetOwner()
+		local inflictor = owner or self.Turret
+		util.BlastDamage(self.Turret, inflictor, tr.HitPos, 400, 100)
+		util.BlastDamage(self.Turret, inflictor, tr.HitPos, 200, 300)
```
Comment (StrawWagen) [inline lua/entities/rail_shell/init.lua:88]: swapped selfTurret and Inflictor here

File: lua/entities/turret_40mm_frag/init.lua [hunk 3]
```diff
@@ -3,106 +3,113 @@ AddCSLuaFile( "shared.lua" )
...
+			effectdata:SetScale( 60 )			// Big splash for big bullets
+			util.Effect( "watersplash", effectdata )
+			self.Entity:Remove()
+			return true
+			
+		end
+		
+		local owner = IsValid( self:GetOwner() ) and self:GetOwner()
+		local inflictor = owner or self.Turret
+		util.BlastDamage(self.Turret, inflictor, self.Entity:GetPos(), 350, 90)
```
Comment (StrawWagen) [inline lua/entities/turret_40mm_frag/init.lua:89]: made damage changes here, and applied blastdamage fix

File: lua/entities/turret_bullets2/shared.lua [hunk 4]
```diff
@@ -116,53 +116,59 @@ function ENT:DoShot()
...
+			
 			self.shootPos:FireBullets({
 				Num=1,
 				Src=self.shootPos:GetPos()+self.shootPos:GetAngles():Up()*10,
 				Dir=self.shootPos:GetAngles():Up()*1,
 				Spread=Vector(0.005,0.005,0),
 				Tracer=0,
 				Force=90,
-				Damage=90,
+				Damage=70,
```
Comment (StrawWagen) [inline lua/entities/turret_bullets2/shared.lua:136]: decreased bullet component of damage to maintain 90 damage on bodyshot

File: lua/entities/turret_bullets2/shared.lua [hunk 5]
```diff
@@ -116,53 +116,59 @@ function ENT:DoShot()
...
+					local tracerEffect=EffectData()
+					tracerEffect:SetStart(self.shootPos:GetPos())
+					tracerEffect:SetOrigin(trace.HitPos)
+					tracerEffect:SetScale(6000)
+					util.Effect("Tracer",tracerEffect)
+					
+					if trace.HitSky then return end
+					
+					local inflictor = self.Shooter or self
+					util.BlastDamage(self, inflictor, trace.HitPos, 90, 20) -- explosion for anti armour power
```
Comment (StrawWagen) [inline lua/entities/turret_bullets2/shared.lua:150]: new blast damage here

File: lua/entities/turret_bullets2/shared.lua [hunk 6]
```diff
@@ -116,53 +116,59 @@ function ENT:DoShot()
...
+					
+					local effectdata = EffectData()
+					effectdata:SetOrigin(trace.HitPos)
+					effectdata:SetScale(1.25) 
+					effectdata:SetRadius( concrete )
+					effectdata:SetNormal( trace.HitNormal )
+					util.Effect("gdcw_universal_impact_t",effectdata)
+					
+					util.Decal("SmallScorch", trace.HitPos + trace.HitNormal, trace.HitPos - trace.HitNormal) -- decal to communicate that yes, this goes boom
```
Comment (StrawWagen) [inline lua/entities/turret_bullets2/shared.lua:159]: decal to help communicate explosion

File: lua/entities/turret_bullets2/shared.lua [hunk 7]
```diff
@@ -116,53 +116,59 @@ function ENT:DoShot()
...
 			self:EmitSound( self.ShotSound, 50, 100 + variance )
 			self:EmitSound( "weapons/ar2/fire1.wav", 70, 60 )
 			
-			
 		end
 		
 		if IsValid(self.shootPos) and SERVER then
 			
+			print( self.Shooter )
+
```
Comment (plally) [inline lua/entities/turret_bullets2/shared.lua:128]: should remove this  print

**Comment by wrefgtzweve** [comment]
can you give me a sign when this gets merged, i'd love to "rework" this pile of shit addon and actually use a base & such

File: lua/entities/turret_bullets2/shared.lua (hunk 7 above)
Comment (StrawWagen) [reply lua/entities/turret_bullets2/shared.lua:128]: i really forgot to remove that, cool, will do

**Pull request merged**